### ISSUE TITLE
Add Feature Selection Algorithms

### ISSUE THREAD

**Issue by cxhernandez** [msmbuilder/msmbuilder#797]
We can start by:

- [ ] porting from [sklearn](http://scikit-learn.org/stable/modules/feature_selection.html)
- [ ] adding a few from [this resource](http://featureselection.asu.edu/algorithms.php) (esp the decision tree-based approaches)
- [ ] implementing [Gradient Boosted Feature Selection (GBFS)](http://alicezheng.org/papers/gbfs.pdf)

cc @msultan

**Comment by msultan** [comment]
i wonder if we need to push all the features through a Normalizer before we can use some of these algorithms. We should be careful about that. The variance threshold based metrics in particular might be troublesome.  

a few technical aspects

1). It would be nice if it could accept a bunch of featurizer objects and return a folder of featurized trajs
2). Some kind of feature descriptors 
3). Our wrapper should hopefully figure out on the fly whether the fully featurized set can fit in memory and if not it should stride appropriately to fit and transform everything later

**Comment by cxhernandez** [comment]
> i wonder if we need to push all the features through a Normalizer before we can use some of these algorithms. We should be careful about that. The variance threshold based metrics in particular might be troublesome.

It appears you're [right](http://stats.stackexchange.com/questions/105478/feature-normalization-standardization-before-or-after-feature-selection)!

> It would be nice if it could accept a bunch of featurizer objects and return a folder of featurized trajs
That's outside the scope of this issue, I think.

> Some kind of feature descriptors 
Agree. We should be able to pass info from `describe_features`.

> Our wrapper should hopefully figure out on the fly whether the fully featurized set can fit in memory and if not it should stride appropriately to fit and transform everything later

I think this is possible, and is done in NumPy at least (but I'm having trouble finding the check rn). Maybe @mpharrigan knows?
